website/content/api-docs/variables.mdx

```diff
@@ -0,0 +1,290 @@
+| `NO`            | `namespace:* variables:write`<br />The read capability on the variable's namespace |
+
+### Parameters
+
+- `namespace` `(string: "default")` - Specifies the variable's namespace. If
+  set, this will override the request body.
+
+- `cas` `(int: <unset>)` - If set, the variable will only be updated if the
+  `cas` value matches the current variables `ModifyIndex`. If the `cas` value is
+  zero, the variable is only created if it does not already exist. This paradigm
+  allows check-and-set style updates.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    -XPUT -d@spec.nsv.json \
+    https://localhost:4646/v1/var/example/first
+```
+
+### Sample Payload
+
+```json
+{
+  "Namespace": "prod",
+  "Path": "example/first",
+  "CreateIndex": 1457,
+  "ModifyIndex": 1457,
+  "CreateTime": 1662061225600373000,
+  "ModifyTime": 1662061225600373000
+  "Items": {
+    "user": "me",
+    "password: "passw0rd1"
+  }
+}
+```
+
+### Sample Response
+
+The response body returns the created or updated variable along with metadata
+created by the server:
+
+```json
+{
+  "Namespace": "prod",
+  "Path": "example/first",
+  "CreateIndex": 1457,
+  "ModifyIndex": 1457,
+  "CreateTime": 1662061225600373000,
+  "ModifyTime": 1662061225600373000,
+  "Items": {
+    "user": "me",
+    "password: "passw0rd1"
+  }
+}
+```
+
+### Sample Response for Conflict
+
+In the case of a compare-and-set conflict, the API will return HTTP error code
+409 and a response body showing the conflicting variable. If the provided ACL
+token does not also have `read` permissions to the variable path, the response
+will include only metadata and not the `Items` field:
+
+```json
+{
+  "Namespace": "prod",
+  "Path": "example/first",
+  "CreateIndex": 1457,
+  "ModifyIndex": 1457,
+  "CreateTime": 1662061225600373000,
+  "ModifyTime": 1662061225600373000,
+  "Items": {
+    "user": "me",
+    "password: "passw0rd1"
+  }
+}
+```
+
+
+## Delete Variable
+
+This endpoint deletes a specific variable by path.
+
+| Method | Path               | Produces           |
+|--------|--------------------|--------------------|
+| `DELETE`  | `/v1/var/:var_path` | `application/json` |
+
+The table below shows this endpoint's support for [blocking queries] and
+[required ACLs].
+
+| Blocking Queries | ACL Required                                                                            |
+|------------------|-----------------------------------------------------------------------------------------|
+| `NO`             | `namespace:* variables:destroy`<br />The destroy capability on the variable's namespace |
+
+### Parameters
+
+- `namespace` `(string: "default")` - Specifies the variable's namespace.
+
+- `cas` `(int: <unset>)` - If set, the variable will only be deleted if the
+  `cas` value matches the current variables `ModifyIndex`. If the value is set
+  to 0, the API will always conflict and return an error if the variable exists.
```

Comment:
@angrycub this description of the behavior of `cas=0` is what this API does currently, but I'm not sure it's actually the _correct_ behavior? What are your thoughts there?

Reply:
I don't think that it's correct.  I've added PR #14555 to update the behavior so that the behavior is more expected.

Reply:
👍 Removed this line.